app/views/kudos/index.html.erb

```diff
@@ -15,6 +16,13 @@
         <td><%= kudo.content %></td>
         <td><%= kudo.giver.email %></td>
         <td><%= kudo.receiver.email %></td>
+        <% if kudo.giver == current_employee %>
```

Comment:
You are checking, if that kudo has been given by currently logged in user, but unfortunately  it it not enough.

Link to edit will look something like this:
`/kudos/1/edit
`
so I can put for example
`/kudos/2/edit`
in my browser, and access the edit page of a kudo, that does not belongs to me - try it out.

Same thing applies to delete - there should be additional check in the controller to make sure, that action can be performed.

Reply:
I increased the level of protection :)